### PR TITLE
Bridge USDC reserves and COMP rewards to Polygon Comet

### PR DESCRIPTION
**Pull request by kevincheng96**
This PR adds ERC20 bridging actions to the proposal for bridging USDC reserves and COMP rewards. I also added support for relaying Polygon token bridging events to our scenarios.

### CODE REVIEW

File: deployments/polygon/usdc/migrations/1675200105_raise_polygon_supply_caps.ts [hunk 1]
```diff
@@ -51,11 +55,53 @@ export default migration('1675200105_raise_polygon_supply_caps', {
       ]
     );
 
+    const RootChainManager = await deploymentManager.existing(
+      'RootChainManager',
+      RootChainManagerAddress
+    );
+    const USDCAmountToBridge = exp(100_000, 6);
+    const COMPAmountToBridge = exp(10_000, 18);
```
Comment (kevincheng96) [inline deployments/polygon/usdc/migrations/1675200105_raise_polygon_supply_caps.ts:63]: Amounts TBD

File: deployments/polygon/usdc/migrations/1675200105_raise_polygon_supply_caps.ts [hunk 2]
```diff
@@ -1,8 +1,12 @@
+import { Contract } from 'ethers';
 import { DeploymentManager } from '../../../../plugins/deployment_manager/DeploymentManager';
 import { migration } from '../../../../plugins/deployment_manager/Migration';
 import { calldata, exp, getConfigurationStruct, proposal } from '../../../../src/deploy';
 import { expect } from 'chai';
 
+const ERC20PredicateAddress = '0x40ec5B33f54e0E8A33A975908C5BA1c14e5BbbDf';
+const RootChainManagerAddress = '0xA0c68C638235ee32657e8f720a23ceC1bFc77C77';
```
Comment (kevincheng96) [inline deployments/polygon/usdc/migrations/1675200105_raise_polygon_supply_caps.ts:8]: Addresses found here: https://static.matic.network/network/mainnet/v1/index.json

File: deployments/polygon/usdc/migrations/1675200105_raise_polygon_supply_caps.ts [hunk 3]
```diff
@@ -51,11 +55,53 @@ export default migration('1675200105_raise_polygon_supply_caps', {
...
+      {
+        contract: COMP,
+        signature: 'approve(address,uint256)',
+        args: [ERC20PredicateAddress, COMPAmountToBridge]
+      },
+      // 5. Bridge COMP from mainnet to Polygon CometRewards using RootChainManager
+      {
+        target: RootChainManager.address,
+        signature: 'depositFor(address,address,bytes)',
+        calldata: depositForCOMPCCalldata
```
Comment (kevincheng96) [inline deployments/polygon/usdc/migrations/1675200105_raise_polygon_supply_caps.ts:104]: Flows documented in Polygon docs: https://wiki.polygon.technology/docs/develop/ethereum-polygon/pos/calling-contracts/erc20/#high-level-flow

Comment (jflatow) [reply deployments/polygon/usdc/migrations/1675200105_raise_polygon_supply_caps.ts:104]: 💪 Definitely agree we want to test this on Goerli/Mumbai

File: scenario/utils/relayPolygonMessage.ts [hunk 4]
```diff
@@ -1,53 +1,134 @@
...
-    });
-
-    setTimeout(() => {
-      reject(new Error('StateSender.StateSynced event listener timed out'));
-    }, EVENT_LISTENER_TIMEOUT);
+  // grab all events on the StateSender contract since the `startingBlockNumber`
+  const stateSyncedEvents: Event[] = [];
+  const filter = stateSender.filters.StateSynced();
+  const logs = await governanceDeploymentManager.hre.ethers.provider.getLogs({
+    fromBlock: startingBlockNumber,
```
Comment (jflatow) [inline scenario/utils/relayPolygonMessage.ts:63]: Nice, this looks better 👍

**Review by jflatow — APPROVED** [review]

File: scenario/utils/relayPolygonMessage.ts [hunk 5]
```diff
@@ -1,53 +1,134 @@
...
-      resolve(log);
-    });
-
-    setTimeout(() => {
-      reject(new Error('StateSender.StateSynced event listener timed out'));
-    }, EVENT_LISTENER_TIMEOUT);
+  // grab all events on the StateSender contract since the `startingBlockNumber`
+  const stateSyncedEvents: Event[] = [];
+  const filter = stateSender.filters.StateSynced();
+  const logs = await governanceDeploymentManager.hre.ethers.provider.getLogs({
```
Comment (scott-silver) [inline scenario/utils/relayPolygonMessage.ts:62]: I think you just rename `logs` to `stateSyncedEvents`, and then delete lines 60 and 68

```
const stateSyncedEvents: Event[]  = await governanceDeploymentManager.hre.ethers.provider.getLogs({
```

File: scenario/utils/relayPolygonMessage.ts [hunk 6]
```diff
@@ -1,53 +1,134 @@
...
+
+export default async function relayPolygonMessage(
   governanceDeploymentManager: DeploymentManager,
   bridgeDeploymentManager: DeploymentManager,
+  startingBlockNumber: number
 ) {
-  const MUMBAI_RECEIVER_ADDRESSS = '0x0000000000000000000000000000000000001001';
-  const EVENT_LISTENER_TIMEOUT = 60000;
+  const POLYGON_RECEIVER_ADDRESSS = '0x0000000000000000000000000000000000001001';
+  const CHILD_CHAIN_MANAGER_PROXY_ADDRESS =
```
Comment (scott-silver) [inline scenario/utils/relayPolygonMessage.ts:41]: super nit, but I believe ALL_CAPS is usually reserved for constants (whereas the value here varies depending on the dm's network)

**Review by scott-silver — APPROVED** [review]
Nice!

File: scenario/utils/relayPolygonMessage.ts (hunk 6 above)
Comment (kevincheng96) [reply scenario/utils/relayPolygonMessage.ts]: Makes sense, done.

File: scenario/utils/relayPolygonMessage.ts (hunk 5 above)
Comment (kevincheng96) [reply scenario/utils/relayPolygonMessage.ts]: Good catch. Done.

File: deployments/polygon/usdc/migrations/1675200105_raise_polygon_supply_caps.ts (hunk 2 above)
Comment (kevincheng96) [reply deployments/polygon/usdc/migrations/1675200105_raise_polygon_supply_caps.ts:8]: Yeah looks like that's the `DummyERC1155` on Polygon. So they probably deployed these two contracts using the same EOA and nonce.

**Pull request merged**